rtos/rtx5/mbed_rtx_conf.h

```diff
@@ -24,6 +24,11 @@
 
 #include "mbed_rtx.h"
 
+/** Any access to RTX5 specific data structures used in common code should be wrapped in ifdef OS_BACKEND_RTX5 */
```

Comment:
I don't get why we have it this way. I would say it's fair enough to try to be 'generic' and let people replace RTX with other RTOSs as long as they implement CMSIS-RTOS2 API. But why do you enable it for CORTEX A and M? In any case that's the only HW we support. We could just define this flag in mbed_rtx_conf.h without checking the cores as it's RTX config and I would expect it should be provided per implementation (together with the storage and handlers files). Also I'm a big fan of mbed flags starting with mbed, so maybe something like MBED_OS_BACKEND_RTX5 for clarity.